test/unittests/OpModel/TTNN/Lib/TestOpModelLib.cpp

```diff
@@ -0,0 +1,660 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "OpModelFixture.h"
+#include "gtest/gtest.h"
+
+#include "ttmlir/OpModel/TTNN/TTNNOpModel.h"
+
+namespace mlir::tt::op_model::ttnn {
+
+class OpModelBase : public OpModelFixture {
+public:
+  mlir::tt::ttnn::TTNNLayoutAttr
+  CreateLayout(llvm::ArrayRef<int64_t> tensorShape,
+               mlir::tt::ttnn::BufferType bufferType,
+               mlir::tt::ttnn::TensorMemoryLayout tensorMemoryLayout,
+               ArrayRef<int64_t> virtualGrid = {8, 8}) {
+    return mlir::tt::ttnn::TTNNLayoutAttr::get(
+        &context, tensorShape, TileType::get(&context, builder.getBF16Type()),
+        bufferType, GridAttr::get(&context, virtualGrid),
+        mlir::tt::ttnn::TensorMemoryLayoutAttr::get(&context,
+                                                    tensorMemoryLayout));
+  }
+};
+
+TEST_F(OpModelBase, ReluInterleaved) {
+  const llvm::SmallVector<int64_t> tensorShape = {64, 1024};
+  const auto workerGrid =
+      CreateWorkerGrid(mlir::tt::ttnn::TensorMemoryLayout::Interleaved, {8, 8});
+  const mlir::tt::ttnn::TTNNLayoutAttr inputLayout_dram =
+      CreateLayout(tensorShape, mlir::tt::ttnn::BufferType::DRAM,
+                   mlir::tt::ttnn::TensorMemoryLayout::Interleaved);
+  const mlir::tt::ttnn::TTNNLayoutAttr inputLayout_l1 =
+      CreateLayout(tensorShape, mlir::tt::ttnn::BufferType::L1,
+                   mlir::tt::ttnn::TensorMemoryLayout::Interleaved);
+
+  std::tuple<bool, std::optional<std::tuple<size_t, size_t, size_t>>,
+             std::optional<std::string>>
+      opConstraint;
+  bool legal = false;
+  std::optional<std::tuple<size_t, size_t, size_t>> l1Usage = std::nullopt;
+  std::optional<std::string> errorMsg = "";
+  size_t cb_size = 0;
+  size_t peak_size = 0;
+  size_t output_size = 0;
+
+  opConstraint = ReluOpInterface::getOpConstraints(
+      tensorShape, inputLayout_dram, tensorShape, inputLayout_dram, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 8192);
+  EXPECT_EQ(output_size, 0);
+  EXPECT_EQ(peak_size, 0);
+
+  opConstraint = ReluOpInterface::getOpConstraints(
+      tensorShape, inputLayout_dram, tensorShape, inputLayout_l1, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 8192);
+  EXPECT_EQ(output_size, 4096);
```

Comment:
x2, double buffered

Reply:
N300 by default has 56 worker cores and 56 interleaved banks. It can have 64 when ethernet cores are used for dispatch.
Thus some cores will have two tiles, and as allocation always goes from bank 0, it's effectively the same as all cores having two tiles.